### PR TITLE
Documenting `Texture`, `TextureView` and `Sampler`.

### PR DESCRIPTION
**Pull request by bzm3r**
# Objective

`Texture` documentation is a little sparse, and it is difficult to say to what extent it can be improved, since it is ultimately a fairly light wrapper around `wgpu`'s `GpuTexture`. Still, improvements could be made by adding relevant links to `Texture` related structures: for example, `Image` and `TextureDescriptor`. 

## Solution

Added a small blurb describing what `Texture`s are used for, and noting its tight connection to `Image`. Also added a sentence noting that creating a texture usually requires setting up a `TextureDescriptor`. Similar light improvements were made to other related data structures.

### CODE REVIEW

**Comment by bzm3r** [comment]
@superdump Curious, as always, to hear your comments, if you have time!

**Review by alice-i-cecile — APPROVED** [review]
Docs look good to me from a writing and teaching perspective.

File: crates/bevy_render/src/render_resource/texture.rs [hunk 1]
```diff
@@ -22,7 +36,7 @@ impl Texture {
         self.id
     }
 
-    /// Creates a view of this texture.
+    /// Creates a [`TextureView`](crate::render_resource::TextureView) of this texture.
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/render_resource/texture.rs:39]:
```suggestion
    /// Creates a [`TextureView`] of this texture.
```
This shouldn't be needed here or elsewhere; the type's already in scope.

File: crates/bevy_render/src/render_resource/texture.rs [hunk 2]
```diff
@@ -67,10 +81,14 @@ pub enum TextureViewValue {
...
-/// Describes a [`Texture`] with its associated metadata required by a pipeline or [`BindGroup`](super::BindGroup).
+/// Describes a [`Texture`] with the associated metadata required by a pipeline or [`BindGroup`](super::BindGroup).
+///
+/// It is created by setting up a [`TextureViewDescriptor`](crate::render_resource::TextureViewDescriptor), and
+/// calling the [`create_view`](crate::render_resource::Texture::create_view) method on a
+/// [`Texture`](crate::render_resource::Texture).
 ///
-/// May be converted from a [`TextureView`](wgpu::TextureView) or [`SurfaceTexture`](wgpu::SurfaceTexture)
-/// or dereferences to a wgpu [`TextureView`](wgpu::TextureView).
+/// Can be converted from [`wgpu::TextureView`](wgpu::TextureView) or wgpu [`wgpu::SurfaceTexture`](wgpu::SurfaceTexture). It can
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/render_resource/texture.rs:90]: Semantic line breaks plz.

File: crates/bevy_render/src/render_resource/texture.rs [hunk 3]
```diff
@@ -7,8 +7,22 @@ pub struct TextureId(Uuid);
...
-/// May be converted from and dereferences to a wgpu [`Texture`](wgpu::Texture).
-/// Can be created via [`RenderDevice::create_texture`](crate::renderer::RenderDevice::create_texture).
+/// Although primarily used to to store data that "texture" the surfaces of solids, textures can also be used as general purpose
+/// storage for data of 1, 2, or 3 dimensional data.
+///
+/// It is created by setting up a [`TextureDescriptor`](crate::render_resource::TextureDescriptor)
+/// and then calling [`RenderDevice::create_texture`](crate::renderer::RenderDevice::create_texture) or  
+/// [`RenderDevice::create_texture_with_data`](crate::renderer::RenderDevice::create_texture_with_data).
+///
+/// Note that a closely related data structure is [`Image`](crate::texture::Image). It can be thought of a CPU-side analogue of
```
Comment (alice-i-cecile) [inline crates/bevy_render/src/render_resource/texture.rs:17]: Semantic line breaks

File: crates/bevy_render/src/render_resource/texture.rs [hunk 4]
```diff
@@ -7,8 +7,22 @@ pub struct TextureId(Uuid);
 
 /// A GPU-accessible texture.
 ///
-/// May be converted from and dereferences to a wgpu [`Texture`](wgpu::Texture).
-/// Can be created via [`RenderDevice::create_texture`](crate::renderer::RenderDevice::create_texture).
+/// Although primarily used to to store data that "texture" the surfaces of solids, textures can also be used as general purpose
```
Comment (DGriffin91) [inline crates/bevy_render/src/render_resource/texture.rs:10]: Maybe something like `Although primarily used to store image data that is applied (mapped) to the surface of polygon`
Using "texture" to help describe what texture is feels a bit funny. Also, there aren't really solids in this context. Faces, fragments, polygons, triangles, etc...

File: crates/bevy_render/src/render_resource/texture.rs [hunk 5]
```diff
@@ -7,8 +7,22 @@ pub struct TextureId(Uuid);
...
+/// It is created by setting up a [`TextureDescriptor`](crate::render_resource::TextureDescriptor)
+/// and then calling [`RenderDevice::create_texture`](crate::renderer::RenderDevice::create_texture) or  
+/// [`RenderDevice::create_texture_with_data`](crate::renderer::RenderDevice::create_texture_with_data).
+///
+/// Note that a closely related data structure is [`Image`](crate::texture::Image). It can be thought of a CPU-side analogue of
+/// [`Texture`](crate::render_resource::Texture), containing the data that will eventually be uploaded to a GPU-side
+/// [`Texture`](crate::render_resource::Texture).
+///
+/// For general information about textures, see the documentation for [`wgpu::Texture`](wgpu::Texture) and the
+/// WGPU specification entry on [`GPUTexture`]. A [`Texture`] may be converted from and dereferences into a
```
Comment (superdump) [inline crates/bevy_render/src/render_resource/texture.rs:22]:
```suggestion
/// WebGPU specification entry on [`GPUTexture`]. A [`Texture`] may be converted from and dereferences into a
```

**Review by superdump — CHANGES_REQUESTED** [review]

File: crates/bevy_render/src/render_resource/texture.rs (hunk 3 above)
Comment (bzm3r) [reply crates/bevy_render/src/render_resource/texture.rs:17]: @alice-i-cecile hmm, can you clarify this?

File: crates/bevy_render/src/render_resource/texture.rs (hunk 4 above)
Comment (bzm3r) [reply crates/bevy_render/src/render_resource/texture.rs]: @DGriffin91 thanks! Rob submitted some suggested changes which I think address this. How do you feel about how it is now?

File: crates/bevy_render/src/render_resource/texture.rs (hunk 3 above)
Comment (alice-i-cecile) [reply crates/bevy_render/src/render_resource/texture.rs:17]: Split your lines at the sentence or clause level when convenient to make them easier to read and review :)

File: crates/bevy_render/src/render_resource/texture.rs (hunk 4 above)
Comment (DGriffin91) [reply crates/bevy_render/src/render_resource/texture.rs]: Yeah, this works fine.

**Pull request closed without merging**